### PR TITLE
Invalidate all cookies after successful password change

### PR DESCRIPTION
**Pull request by xImouto**

### CODE REVIEW

File: packages/server/src/middlewares/authorization.ts [hunk 1]
```diff
@@ -17,15 +21,23 @@ export const isAuthenticated = async (
...
+    const userFromDb = (await prisma.user.findUnique({
       where: { id },
       include: {
         profile: true
       }
-    }) as UserWithHashedPassAndProfile;
+    })) as UserWithHashedPassAndProfile;
     if (!userFromDb) throw new NotAuthenticatedError();
+
+    if (userFromDb.tokenVersion > tokenVersion) {
```
Comment (Devorein) [inline packages/server/src/middlewares/authorization.ts:37]: This should check for equality

**Review by Devorein — CHANGES_REQUESTED** [review]

File: packages/server/src/models/User.ts [hunk 2]
```diff
@@ -139,12 +139,17 @@ export async function getUserByUsernameOrId(
...
         id
       },
-      data: { hashedPass }
+      data: {
+        hashedPass,
+        passwordUpdatedAt: new Date()
+      }
     });
+    // for testing purposes
+    return removeHashedPass(user);
```
[the remark this answers is not in the document]
Comment (xImouto) [reply packages/server/src/models/User.ts]: done in latest commit

File: packages/server/src/middlewares/authorization.ts (hunk 1 above)
Comment (xImouto) [reply packages/server/src/middlewares/authorization.ts:37]: This will throw error if they are both equal. I dont think that should be the case.

Comment (Devorein) [reply packages/server/src/middlewares/authorization.ts:37]: Sorry I meant it should throw error if they are not equal.

**Review by Devorein — APPROVED** [review]

**Pull request merged**